services/crawler/crawler.go

```diff
@@ -0,0 +1,104 @@
+package crawler
+
+import (
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/beego/beego/v2/core/logs"
+)
+
+const googleSearchBaseUrl = "https://www.google.com/search"
+
+var doc *goquery.Document
+var htmlResponse string
+var selectorMapping = map[string]string{
+	"topAdWordAdvertisersCount":   "#tads .uEierd",
+	"topAdWordAdvertisersUrls":    "#tads .Krnil",
+	"totalAdWordAdvertisersCount": ".Krnil",
+	"resultsCount":                "#rso .yuRUbf",
+	"resultsUrls":                 "#rso .yuRUbf > a",
+	"totalLinksCount":             "a",
+}
+
+type CrawlData struct {
+	TopAdWordAdvertisersCount   int
+	TopAdWordAdvertisersUrls    []string
+	TotalAdWordAdvertisersCount int
+	ResultsCount                int
+	ResultsUrls                 []string
+	TotalLinksCount             int
+	Html                        string
+}
+
+// Call this function with your search key and base url, it will return the necessary crawled data
+// Ex: Crawl("Buy laptop", "baseUrl")
```

Comment:
Just minor issue here as the doc block has not been updated to reflect the removal of the second parameter for `baseUrl`

```suggestion
// Call this function with your search key, it will return the necessary crawled data
// Ex: Crawl("Buy laptop")
```